src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3838,17 +3838,31 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                 type = null;
                 name = this.EatToken(SyntaxKind.ArgListKeyword);
             }
-
-            EqualsValueClauseSyntax def = null;
+            SyntaxToken exclamation = null;
+            SyntaxToken equals = null;
+            if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+            {
+                exclamation = this.EatToken(SyntaxKind.ExclamationToken);
+            }
+            else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
+            {
+                var notEq = this.EatToken(SyntaxKind.ExclamationEqualsToken);
+                equals = ConvertToMissingWithTrailingTrivia(notEq, SyntaxKind.EqualsToken);
+                equals = AddError(equals, ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals);
+                exclamation = SyntaxFactory.MissingToken(SyntaxKind.ExclamationToken);
```

Comment:
Don't we need to take any leading trivia from the ExclamationEqualsToken and add it to this exclamation token?